### DISCUSSION TITLE
useAsyncQuery runs on client-side too?

### DISCUSSION THREAD

**Discussion by jwhscholten** [nuxt-community/apollo-module#503] [Q&A]
I'm using useAsyncQuery to fetch my data. I was expecting this to run on the server only. During navigation I see my requests running client side as well, exposing my access token to the public. What's the correct approach to hide these requests from the client?

**Comment by RihanArfan** [comment]
You could use server only components 

https://nuxt.com/docs/guide/directory-structure/components#server-components
